spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-cloudfoundry/src/main/java/org/springframework/cloud/data/module/deployer/cloudfoundry/AbstractPaginatedResponse.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.module.deployer.cloudfoundry;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+abstract class AbstractPaginatedResponse {
```

Comment:
@author tags going in.  Javadoc will take a bit longer, but I will make a start on it. There are only a few places it is really necessary: all of the public interfaces, for example, and some of the internal classes that aren't obvious.  Most of the Request/Response pairs are self-explanatory, for example (deliberately mirroring the REST API doc).